### PR TITLE
Adding getDividends and some other small changes

### PR DESCRIPTION
**Pull request by 42iscool42**
The most obvious thing is that I implemented the getDividends function in IEX.h.  I also renamed all variables in the data structs that are being used to follow the same convention.  In addition, I made a private namespace for some of the functions that were being used in multiple places and I moved all the functions to the Stock class.  I did this because it is organized this way in the documentation for the web API and I felt the multiple classes that were there seemed arbitrary.  Now, there is Stock::getPrice(), Stock::getStats(), etc.

### CODE REVIEW

File: IEX.h [hunk 1]
```diff
@@ -85,310 +86,353 @@ namespace IEX {
...
-                pd.stock_symbol = stock_symbol;
-                pd.latest_price = jsonData.asDouble();
+                dividend.flag = getString(dividend_json["flag"]);
 
-                return pd;
+                dividend.type = getString(dividend_json["type"]);
+
+                dividend.qualified = getString(dividend_json["qualified"]);
+
+                std::string indicated = getString(dividend_json["indicated"]);
```
Comment (mpherg) [inline IEX.h]: Why not just call `getDouble` here? Or add a `getFloat`?

File: IEX.h [hunk 2]
```diff
@@ -85,310 +86,353 @@ namespace IEX {
...
+        //  stock_symbol - the stock symbol you want to query
+        //  time_period - the time period you want to look in.  Allowed
+        //  values are: 5y, 2y, 1y, ytd, 6m, 3m, and 1m
+        //
+        // Output:
+        //  Will returns a DividendsData struct with the relevant info
+        //  on success or an empty struct with the success field set to
+        //  false
+        static DividendsData getDividends(const std::string stock_symbol,
+                                         const std::string time_period) {
```
Comment (mpherg) [inline IEX.h]: Since this time period is well defined, I think this should be an enumeration so as to avoid any confusion if the wrong string is passed.

**Comment by 42iscool42** [comment]
I would agree with you, but we can't have enum values that start with numbers so 2y or 3m wouldn't be a valid enum value.

File: IEX.h (hunk 1 above)
Comment (42iscool42) [reply IEX.h]: I had it set up differently earlier when developing. I changed it and forgot to make that change.

File: IEX.h (hunk 2 above)
Comment (42iscool42) [reply IEX.h]: I would agree with you, but we can't have enum values that start with numbers so 2y or 3m wouldn't be a valid enum value.

Comment (rodolfobandeira) [reply IEX.h]: We could add a "parameter validator" for this to make sure only valid methods are passed or just adding an error handler catching the HTTP 400 (bad request) exceptions coming from the server side. Doesn't need to be implemented now, we can just create an issue as "things to improve in the future".

File: IEX.h (hunk 1 above)
Comment (42iscool42) [reply IEX.h]: Actually, on further review, I remembered that indicated is sometimes just an empty string and not a number, so this part is necessary.

File: IEX.h (hunk 2 above)
Comment (42iscool42) [reply IEX.h]: I think the way I have it set up is more appropriate.  For me, having a string option is easier to use than using an enum.

**Comment by rodolfobandeira** [comment]
Hey @42iscool42, any update on this? I can help out fixing these conflicts if you're busy. (I could fix it and make a PR in your fork/branch)

Thanks

**Comment by 42iscool42** [comment]
Sorry, I made the changes I just merged the conflicts wrong and so the new pull request did not get done correctly.  The pull request should now have getCompanyLogo

**Comment by 42iscool42** [comment]
Small problem with the merge has now been fixed hopefully.  Everything works.  Sorry about the issues with this, I'm fairly new to pull requests on Github

**Comment by rodolfobandeira** [comment]
Hey @42iscool42 

That was fast! I'm gonna merge it even though now we're not handling when there are no arguments but I'm gonna try fix that.

Thank you very much!

```
./compile.sh 
Starting compilation
Done
./main 
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
Aborted (core dumped)
```

**Pull request merged**